src/Http/Routing/src/Matching/CandidateSet.cs

```diff
@@ -142,8 +150,15 @@ public void SetValidity(int index, bool value)
                 ThrowIndexArgumentOutOfRangeException();
             }
 
-            ref var original = ref _candidates[index];
-            _candidates[index] = new CandidateState(original.Endpoint, original.Values, original.Score >= 0 ^ value ? ~original.Score : original.Score);
+            ref var original = ref Candidates[index];
```

Comment:
The pattern of all of these methods now is that there's an instance method that turns an index into a `ref CandidateState`, and a static one with all of the logic.